aiosip/peers.py

```diff
@@ -135,11 +135,17 @@ def _make_key(self, protocol):
         peer_addr = protocol.transport.get_extra_info('peername')
         return local_addr, peer_addr
 
-    def _create_server(self, local_addr):
-        return self._loop.create_server(
-            lambda: TCP(app=self._app, loop=self._loop),
-            host=local_addr[0],
-            port=local_addr[1])
+    def _create_server(self, local_addr, sock):
+
+        if local_addr:
+            return self._loop.create_server(
+                lambda: TCP(app=self._app, loop=self._loop),
+                host=local_addr[0],
+                port=local_addr[1])
+        else:
+            return self._loop.create_server(
+                lambda: TCP(app=self._app, loop=self._loop),
+                sock=sock)
```

Comment:
Even it was a criticism, you have the right to have your personal opinions ;-)
More seriously, we use that because we combine aiosip with API-Hour, it's a gunicorn worker for any AsyncIO server.
With that, we can make easily daemons that are multi-workers, to bypass GIL and be multi-CPU.
to do that, it must be gunicorn that opens the socket, before to fork the process.
The socket is passed after the fork to the API-Hour container to enable you to plug with your AsyncIO server.

In case of you are interested in, we can publish an example, based from our actual production.

Have a nice weekend.

Reply:
Oh wow that sounds useful. I didn't know there was such a thing, but it makes plenty sense. I guess another use case would be just plain old socket activation style service management tool.

Looked that up and going to play with it. Nice work.